LoM-Install/arista/do-install.py

```diff
@@ -0,0 +1,768 @@
+'''
+
+
+
+'''
+
+from __future__ import print_function  # Python 2/3 compatibility
+import os
+import subprocess
+import signal
+import time
+import json
+import sys
+import argparse
+import shutil
+import logging
+
+import arista_eapi_helper as eapi_helper
+import arista_cli_helper as cli_helper
+from common import *
+
+'''
+Sample lom Installer code after extracking to default directory.
+
+/tmp/lom-selfextract.lMQSUE/
+|-- config
+|   |-- actions.conf.json
+|   |-- bindings.conf.json
+|   |-- globals.conf.json
+|   |-- procs.conf.json
+|   `-- security
+|       |-- client_certs
+|       |   |-- client.cer
+|       |   |-- client.csr
+|       |   `-- client.key
+|       |-- dsmsroot.cer
+|       |-- dsmsroot.key
+|       |-- openssl.cnf
+|       |-- openssl_ca.cnf
+|       |-- server.csr
+|       |-- streamingtelemetryserver.cer
+|       `-- streamingtelemetryserver.key
+|-- install
+|   |-- LoM-install.sh
+|   |-- bin
+|   |   |-- LoMCli
+|   |   |-- LoMEngine
+|   |   `-- LoMPluginMgr
+|   `-- startup
+|       |-- arista_cli_helper.py
+|       |-- arista_eapi_helper.py
+|       |-- cli_tools.py
+|       |-- common.py
+|       `-- do-install.py
+`-- libs
+    |-- libsodium.so
+    |-- libsodium.so.26
+    |-- libsodium.so.26.1.0
+    |-- libzmq.so
+    |-- libzmq.so.5
+    `-- libzmq.so.5.2.6
+'''
+
+# Global variables
+g_installer_dir = None
+g_lom_engine_path = None
+g_lom_plugin_mgr_path = None
+g_lom_cli_path = None
+g_config_dir = None
+g_switch_cli_handler = None
+g_switch_eapi_handler = None
+g_procs_keys = None
+g_previous_config = None
+
+def setup(installation_dir):
+    """
+    Sets up the installation environment.
+
+    This function initializes global variables, writes chip details to a file,
+    enables Unix eAPI protocol, and creates instances of AristaSwitchCLIHelper
+    and AristaSwitchEAPIHelper.
+
+    Parameters:
+    installation_dir (str): The directory where the installation files are located.
+
+    Returns:
+    bool: True if the setup was successful, False otherwise.
+    dict: A dictionary containing the existing installation configuration.
+
+    Raises:
+    Exception: If there is an error during setup.
+    """
+    global g_lom_engine_path, g_lom_plugin_mgr_path, g_lom_cli_path, g_config_dir, g_procs_keys, g_installer_dir, \
+            g_switch_cli_handler, g_switch_eapi_handler
+    existing_install_config = {}
+
+    # Get the lom extraction directory path e.g. /mnt/flash/lom/active
+    g_installer_dir = installation_dir
+    logger.log("Working Installation directory: {}".format(g_installer_dir))
```

Comment:
Add To-Do comment : This is a shared signature across all vendoes. Must consult team before changing.

Reply:
added